src/app/util/ClientMonitoringRegistrationTable.cpp

```diff
@@ -0,0 +1,37 @@
+/**
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "ClientMonitoringRegistrationTable.h"
+
+namespace chip {
+
+/**********************************************************
+ * Attributes Definition
+ *********************************************************/
+
+ClientMonitoringRegistrationTable ClientMonitoringRegistrationTable::sInstance;
+
+/**********************************************************
+ * ClientMonitoringRegistrationTable Implementation
+ *********************************************************/
+
+ClientMonitoringRegistrationTable & ClientMonitoringRegistrationTable::Instance()
+{
+    return sInstance;
+}
```

Comment:
We should not use a singleton for this, if it all possible. If there is a single instance in the short term, so be it, but making more and more singletons is unfortunate as it prevents scaling for tests.

Reply:
Replaced the singleton pattern with a version of the factory design pattern. You create a registration table every time you need to access the nvm data and the object gets deleted when you are done using it.